sdk/samples/iot/paho_iot_hub_pnp_component_sample.c

```diff
@@ -963,45 +963,45 @@ static void handle_command_request(
   az_iot_status status = AZ_IOT_STATUS_UNKNOWN;
 
   // Invoke command and retrieve status and response payload to send to server.
-  if (az_span_is_content_equal(thermostat_1.component_name, command_request->component_name))
+  if (az_span_size(command_request->component_name) == 0)
   {
-    if (pnp_thermostat_process_command_request(
-            &thermostat_1,
+    if (az_result_succeeded(temp_controller_process_command_request(
             command_request->command_name,
             message_span,
             publish_message.payload,
             &publish_message.out_payload,
-            &status))
+            &status)))
     {
       IOT_SAMPLE_LOG_AZ_SPAN(
-          "Client invoked command on Temperature Sensor 1:", command_request->command_name);
+          "Client invoked command on Temperature Controller:", command_request->command_name);
     }
   }
-  else if (az_span_is_content_equal(thermostat_2.component_name, command_request->component_name))
+  else if (az_span_is_content_equal(thermostat_1.component_name, command_request->component_name))
```

Comment:
I think there will be some cases where `is_component_in_model` will return true because of the `az_json_token_is_text_equal` comparison, but then this byte-by-byte comparison will unexpectedly return false. Can you try out a component name with some escaped characters within the JSON, while leaving the component names in the array, unescaped?

Reply:
These values are not from JSON. The `thermostat` is a struct with a passed component name and the `command_request` is the parsed command request from an MQTT topic.